core/trino-main/src/main/java/io/trino/execution/SqlQueryExecution.java

```diff
@@ -530,13 +516,6 @@ private void planDistribution(PlanRoot plan)
                 taskDescriptorStorage);
 
         queryScheduler.set(scheduler);
-
-        // if query was canceled during scheduler creation, abort the scheduler
-        // directly since the callback may have already fired
-        if (stateMachine.isDone()) {
-            scheduler.abort();
-            queryScheduler.set(null);
```

Comment:
Shouldn't `queryScheduler.set(null);` be called anyway (e.g. to free resources)?

Reply:
The `queryScheduler` variable is nullified in `buildQueryInfo` after building a final query info. It is currently not nullified in case of a failure or completion anywhere else, thus it seemed to be weird to nullify it here.

Reply:
> It is currently not nullified in case of a failure or completion anywhere else, thus it seemed to be weird to nullify it here.

So was this a race between `buildQueryInfo` and nullification here?

Reply:
I doubt there any. It looked very weird. Later on there's another check where the scheduler is not getting nullified: https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/execution/SqlQueryExecution.java#L423.